src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs

```diff
@@ -123,24 +131,39 @@ internal bool ShouldMonitorHostHealth
 
         public async Task StartAsync(CancellationToken cancellationToken)
         {
-            _startupLoopTokenSource = new CancellationTokenSource();
-            var startupLoopToken = _startupLoopTokenSource.Token;
-            var tokenSource = CancellationTokenSource.CreateLinkedTokenSource(startupLoopToken, cancellationToken);
+            // Shutdown if RunFromZipFailed
```

Comment:
nit: move this comment and the formation of the path into the CheckFileSystem method. That way if we need other checks in the future, they're just added inside that method.